src/main/kotlin/no/nav/familie/ef/sak/vilkår/VurderingService.kt

```diff
@@ -166,6 +170,102 @@ class VurderingService(
         tilbakestillEndretTidForKopierteVurderinger(kopiAvVurderinger, tidligereVurderinger)
     }
 
+    @Transactional
+    fun gjenbrukInngangsvilkårVurderinger(
+        nåværendeBehandlingId: UUID,
+        tidligereBehandlingId: UUID,
+    ) {
+        validerBehandlingForGjenbruk(
+            nåværendeBehandlingId,
+            tidligereBehandlingId
+        )
+        val forrigeVurderingIdTilNåværendeBarnMap =
+            finnBarnPåBeggeBehandlinger(tidligereBehandlingId, nåværendeBehandlingId)
+        val sivilstandErLik =
+            hentGrunnlagsdataOgValiderSivilstand(tidligereBehandlingId, nåværendeBehandlingId)
+        val (tidligereVurderinger, nåværendeVurderinger) = finnVurderingerSomSkalGjenbrukes(
+            tidligereBehandlingId,
+            sivilstandErLik,
+            forrigeVurderingIdTilNåværendeBarnMap,
+            nåværendeBehandlingId
+        )
+
+        val vurderingerSomSkalLagres = tidligereVurderinger.mapNotNull { tidligereVurdering ->
+            if (tidligereVurdering.type == VilkårType.ALENEOMSORG) {
+                val barnForVurdering = forrigeVurderingIdTilNåværendeBarnMap[tidligereVurdering.barnId]
+                val nåværendeVurdering =
+                    nåværendeVurderinger.firstOrNull { nåværendeVurdering -> nåværendeVurdering.type == tidligereVurdering.type  && nåværendeVurdering.barnId == barnForVurdering?.id }
+                if (nåværendeVurdering != null) {
+                    tidligereVurdering.copy(
+                        id = nåværendeVurdering.id,
+                        behandlingId = nåværendeBehandlingId,
+                        sporbar = nåværendeVurdering.sporbar,
+                        barnId = nåværendeVurdering.barnId,
+                    )
+                } else {
+                    null
+                }
+            } else {
+                val nåværendeVurdering = nåværendeVurderinger.first { b -> tidligereVurdering.type == b.type }
+                tidligereVurdering.copy(
+                    id = nåværendeVurdering.id,
+                    behandlingId = nåværendeBehandlingId,
+                    sporbar = nåværendeVurdering.sporbar,
+                )
+            }
+        }
+        vilkårsvurderingRepository.updateAll(vurderingerSomSkalLagres)
+    }
+
+    private fun finnBarnPåBeggeBehandlinger(
+        tidligereBehandlingId: UUID,
+        nåværendeBehandlingId: UUID
```

Comment:
Det kan være fint å ha argument i den samme ordningen, dvs nåværende, og tidligere for hver metode

Reply:
enig, byttet om på rekkefølger i [c593a9d](https://github.com/navikt/familie-ef-sak/pull/1740/commits/c593a9dea7933755dfa27eccbbbd10521a4ac189)